src/Components/Aspire.Azure.AI.OpenAI/AspireAzureOpenAIChatClientExtensions.cs

```diff
@@ -0,0 +1,84 @@
+{
+    private const string DeployentKey = "Deployment";
+    private const string ModelKey = "Model";
+
+    /// <summary>
+    /// Registers a singleton <see cref="IChatClient"/> in the services provided by the <paramref name="builder"/>.
+    ///
+    /// Additionally, registers the underlying <see cref="AzureOpenAIClient"/> and <see cref="OpenAIClient"/> as singleton services.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configurePipeline">An optional method that can be used for customizing the <see cref="IChatClient"/> pipeline.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="AzureOpenAISettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureClientBuilder">An optional method that can be used for customizing the <see cref="IAzureClientBuilder{AzureOpenAIClient, AzureOpenAIClientOptions}"/>.</param>
+    /// <param name="deploymentName">Optionally specifies the deployment name. If not specified, a value will be taken from the connection string.</param>
+    /// <remarks>Reads the configuration from "Aspire.Azure.AI.OpenAI" section.</remarks>
+    public static void AddAzureOpenAIChatClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Func<ChatClientBuilder, ChatClientBuilder>? configurePipeline = null,
+        Action<AzureOpenAISettings>? configureSettings = null,
+        Action<IAzureClientBuilder<AzureOpenAIClient, AzureOpenAIClientOptions>>? configureClientBuilder = null,
+        string? deploymentName = null)
+    {
+        builder.AddAzureOpenAIClient(connectionName, configureSettings, configureClientBuilder);
+
+        builder.Services.AddSingleton(services =>
+        {
+            var chatClientBuilder = new ChatClientBuilder(services);
+            configurePipeline?.Invoke(chatClientBuilder);
+
+            deploymentName ??= GetRequiredDeploymentName(builder.Configuration, connectionName);
+
+            var innerClient = chatClientBuilder.Services
+                .GetRequiredService<AzureOpenAIClient>()
+                .AsChatClient(deploymentName);
+
+            return chatClientBuilder.Use(innerClient);
+        });
+    }
+
+    private static string GetRequiredDeploymentName(IConfiguration configuration, string connectionName)
+    {
+        string? deploymentName = null;
+
+        if (configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            var connectionBuilder = new DbConnectionStringBuilder { ConnectionString = connectionString };
+            deploymentName = (connectionBuilder[DeployentKey] ?? connectionBuilder[ModelKey])?.ToString();
```

Comment:
It looks like DbConnectionStringBuilder's indexer actually throws rather than returning null if the keyword isn't found. Should these instead use TryGetValue?

Reply:
https://github.com/dotnet/runtime/blob/e32148a8bd322f0b7cd5368d4c3810354d9fc14a/src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionStringBuilder.cs#L78-L90

Reply:
Example: we use `connectionBuilder.TryGetValue(ConnectionStringEndpoint, out var endpoint)` in `AddOpenAIClientFromConfiguration`.